src/arm/phyboard.c

```diff
@@ -0,0 +1,672 @@
+    b->pins[37].gpio.parent_id = 0;
+    b->pins[37].gpio.mux_total = 0;
+
+    strncpy(b->pins[38].name, "X_GPIO0_7", MRAA_PIN_NAME_SIZE);
+    b->pins[38].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 0 };
+    b->pins[38].gpio.pinmap = (0*32 + 7);
+    b->pins[38].gpio.parent_id = 0;
+    b->pins[38].gpio.mux_total = 0;
+
+    strncpy(b->pins[39].name, "X_AM335_EXT_WAKEUP", MRAA_PIN_NAME_SIZE);
+    b->pins[39].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    strncpy(b->pins[40].name, "X_INT_RTCn", MRAA_PIN_NAME_SIZE);
+    b->pins[40].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    strncpy(b->pins[41].name, "GND", MRAA_PIN_NAME_SIZE);
+    b->pins[41].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    /* PIN-Number "42" with name "X_GPIO3_7_nPMOD_PW RFAIL" */
+    strncpy(b->pins[42].name, "X_GPIO3_7", MRAA_PIN_NAME_SIZE);
+    b->pins[42].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 0 };
+    b->pins[42].gpio.pinmap = (3*32 + 7);
+    b->pins[42].gpio.parent_id = 0;
+    b->pins[42].gpio.mux_total = 0;
+
+    strncpy(b->pins[43].name, "nRESET_IN", MRAA_PIN_NAME_SIZE);
+    b->pins[43].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    strncpy(b->pins[44].name, "X_GPIO1_31", MRAA_PIN_NAME_SIZE);
+    b->pins[44].capabilities = (mraa_pincapabilities_t){ 1, 1, 0, 0, 0, 0, 0, 0 };
+    b->pins[44].gpio.pinmap = (1*32 + 31);
+    b->pins[44].gpio.parent_id = 0;
+    b->pins[44].gpio.mux_total = 0;
+
+    strncpy(b->pins[45].name, "X_AM335_NMIn", MRAA_PIN_NAME_SIZE);
+    b->pins[45].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    strncpy(b->pins[46].name, "GND", MRAA_PIN_NAME_SIZE);
+    b->pins[46].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    // NOT DONE / TESTED YET
+    strncpy(b->pins[47].name, "X_AIN4", MRAA_PIN_NAME_SIZE);
+    b->pins[47].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 1, 0 };
+
+    // NOT DONE / TESTED YET
+    strncpy(b->pins[48].name, "X_AIN5", MRAA_PIN_NAME_SIZE);
+    b->pins[48].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 1, 0 };
+
+    // NOT DONE / TESTED YET
+    strncpy(b->pins[49].name, "X_AIN6", MRAA_PIN_NAME_SIZE);
+    b->pins[49].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 1, 0 };
+
+    // NOT DONE / TESTED YET
+    strncpy(b->pins[50].name, "X_AIN7", MRAA_PIN_NAME_SIZE);
+    b->pins[50].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 1, 0 };
+
+    strncpy(b->pins[51].name, "GND", MRAA_PIN_NAME_SIZE);
+    b->pins[51].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    /* to CHECK X_GPIO_CKSYNC */
+    strncpy(b->pins[52].name, "X_GPIO_CKSYNC", MRAA_PIN_NAME_SIZE);
+    b->pins[52].capabilities = (mraa_pincapabilities_t){ 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    strncpy(b->pins[53].name, "X_USB_ID_EXP", MRAA_PIN_NAME_SIZE);
```

Comment:
`MRAA_PIN_NAME_SIZE` is 12 and `b->pins[53].name` is allocated to 12 symbols, that means the `strncpy()` will not copy the null-terminator of the string and it won't fit anyway - please consider shortening the name.